packages/mongo/mongo_driver.js

```diff
@@ -131,14 +131,17 @@ MongoConnection = function (url, options) {
   self._onFailoverHook = new Hook;
 
   var mongoOptions = _.extend({
-      db: { safe: true },
-      // Set reconnectTries to 0 which means keep trying to reconnect forever,
-      // rather than the default of losing the connection permanently after 30
-      // retries (separated by 1000ms).
-      server: { reconnectTries: 0 },
-      replSet: {}
+    db: { safe: true },
+    // http://mongodb.github.io/node-mongodb-native/2.2/api/Server.html
+    server: {
+      // Reconnect on error.
+      autoReconnect: true,
```

Comment:
I'm not sure this is actually an option any more. Well, actually the documentation is inconsistent.

Here: http://mongodb.github.io/node-mongodb-native/2.2/reference/connecting/connection-settings/ it definitely mentions `autoReconnect` in the same breath as `reconnectTries` although doesn't actually mention that either should be on the `server` key.

Here: `http://mongodb.github.io/node-mongodb-native/2.2/api/Server.html` it doesn't mention `autoReconnect`.

Ok, I see there is something in the code: https://github.com/mongodb/node-mongodb-native/blob/2.2/lib/server.js#L73